### PR TITLE
[wip] store receipts: step 2 

### PR DESCRIPTION
**Pull request by AskAlexSharov**
working on data correctness...

### CODE REVIEW

File: core/rawdb/accessors_canonical_reader.go [hunk 1]
```diff
@@ -93,7 +99,8 @@ func (*CanonicalReader) BaseTxnID(tx kv.Tx, blockNum uint64, blockHash common2.H
 		}
 		return kv.TxnId(_min), nil
 	}
-	return kv.TxnId(b.BaseTxnID), nil
+	return kv.TxnId(b.BaseTxnID + 1), nil
```
Comment (JkLondon) [inline core/rawdb/accessors_canonical_reader.go:102]: what was the purpose?

File: eth/stagedsync/stage_custom_trace.go [hunk 2]
```diff
@@ -73,114 +84,110 @@ func SpawnCustomTrace(s *StageState, txc wrap.TxContainer, cfg CustomTraceCfg, c
...
+	var lastBlockNum uint64
+
+	canonicalReader := rawdb.NewCanonicalReader()
+	lastFrozenID, err := canonicalReader.LastFrozenTxNum(tx)
+	if err != nil {
+		return err
+	}
+
+	var baseBlockTxnID, txnID kv.TxnId
+	fmt.Printf("dbg1: %s\n", tx.ViewID())
```
Comment (JkLondon) [inline eth/stagedsync/stage_custom_trace.go:132]: fix printf

File: turbo/jsonrpc/eth_receipts.go [hunk 3]
```diff
@@ -335,6 +337,12 @@ func (api *APIImpl) getLogsV3(ctx context.Context, tx kv.TemporalTx, begin, end
 				continue
 			}
 			blockHash = header.Hash()
+
+			rrrr := rawdb.NewCanonicalReader()
```
Comment (JkLondon) [inline turbo/jsonrpc/eth_receipts.go:341]: why this strange naming?)

Comment (AskAlexSharov) [reply turbo/jsonrpc/eth_receipts.go:341]: non-prod-ready PR yet

**Comment by Giulio2002** [comment]
Is this ready?

**Pull request closed without merging**